spanner/spanner_snippets/snippet.go

```diff
@@ -109,13 +117,29 @@ var (
 		"createtablewithdatatypes":        createTableWithDatatypes,
 		"createtabledocswithtimestamp":    createTableDocumentsWithTimestamp,
 		"createtabledocswithhistorytable": createTableDocumentsWithHistoryTable,
+		"listbackupoperations":            listBackupOperations,
+		"listdatabaseoperations":          listDatabaseOperations,
+	}
+
+	backupCommands = map[string]backupCommand{
+		"createbackup":  createBackup,
+		"cancelbackup":  cancelBackup,
+		"listbackups":   listBackups,
+		"updatebackup":  updateBackup,
+		"deletebackup":  deleteBackup,
+		"restorebackup": restoreBackup,
 	}
 )
 
+var (
+	validDBPattern = regexp.MustCompile("^(.*)/databases/(.*)$")
+)
+
 // [START spanner_create_database]
 
-func createDatabase(ctx context.Context, w io.Writer, adminClient *database.DatabaseAdminClient, db string) error {
-	matches := regexp.MustCompile("^(.*)/databases/(.*)$").FindStringSubmatch(db)
+func createDatabase(ctx context.Context, w io.Writer, adminClient *spadmindb.DatabaseAdminClient, db string) error {
+	// validDBPattern is defined as regexp.MustCompile("^(.*)/databases/(.*)$")
```

Comment:
Can we move the declaration of `validDBPattern` inside the snippet, outside this function? Then we can delete this comment.

Reply:
There are multiple snippets using this `validDBPattern`. I can't just move it to this snippet.

Reply:
Yeah I think we need to move this back into the function. I agree that on the main implementation we decided to move it out but in the case of samples, customers only see what's in context of the function so it has to be redefined in each function.

We also shouldn't change existing samples because that means the docs team has to regenerate the docs.

Reply:
Yeah, changed back to the original version.